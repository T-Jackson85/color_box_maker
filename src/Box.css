.Box-Div {
    margin: auto;
    margin-top: 20px;
    margin-bottom: 2px;
  }
  
  .Box-Button {
    border: 1px solid black;
    border-radius: 5px;
  }
  
  .Box-Button:hover {
    cursor: pointer;
    border: 2px solid black;
  }
