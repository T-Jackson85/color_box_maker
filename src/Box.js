import React from "react";
import './Box.css'

function Box({
  id, 
  height=4, 
  width=4, 
  backgroundColor="green", 
  handleDelete
}) {
  const deleteBox = () => handleDelete(id);

  return (
    <>
      <div style={{height: `${height}em`, 
                   width: `${width}em`, 
                   backgroundColor
           }}
           className="Box-Div"
      />
      <button className="Box-Button" style={{width: `${width}em`, backgroundColor}} onClick={deleteBox}>X</button>
    </>
  );
}

export default Box