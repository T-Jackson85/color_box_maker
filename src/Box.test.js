import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

it("renders Box (smoke test)", function() {
  render(<Box />);
});

it("snapshot test of Box", function() {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});