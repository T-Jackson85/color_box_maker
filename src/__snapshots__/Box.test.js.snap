// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test of Box 1`] = `
<DocumentFragment>
  <div
    class="Box-Div"
    style="height: 4em; width: 4em; background-color: green;"
  />
  <button
    class="Box-Button"
    style="width: 4em; background-color: green;"
  >
    X
  </button>
</DocumentFragment>
`;
